### PR TITLE
Page: History Page Added Ft. HistoryAPI Func Added

### PR DESCRIPTION
**Pull request by manojsarna**
### Features Added 
- [x] History Page Added with History API & Functionality ,Count on Icons
- [x] Toast Added with Toast Stacking in Toast Container

Deploy Link to Check : https://incomparable-gnome-0857d1.netlify.app/

### Files To Check

[/history-context.js](https://github.com/manojsarna/manoj-smashtube-video-lib-react/compare/history-page-API-functionality?expand=1#diff-a6c8a931755217e4bfce18c9f4f01d933a7b95472df4737545b8dec3f99080a4)

[History.jsx](https://github.com/manojsarna/manoj-smashtube-video-lib-react/compare/history-page-API-functionality?expand=1#diff-25403c5ee65592b97b6de38d14b2103296d363142920966d0ebc0d94121f3fcc)

### Ignore Rest of the Files

image

### CODE REVIEW

File: src/context/history-context/history-context.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+  useEffect(() => {
+    (async function () {
+      if (encodedToken) {
+        try {
+          const historyResponse = await axios.get("/api/user/history", {
+            headers: {
+              authorization: encodedToken,
+            },
+          });
```
Comment (m26unkwn) [inline src/context/history-context/history-context.js:23]: I would suggest that try to do it like this. ` useEffect(()=> {
if(encodedToken){
then your asncy func
}
},[dep])`

In your case you are running the function and then checking whether the encoded token is there or not.

File: src/context/history-context/history-context.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  //       return prev;
+  //     } else {
+  //       return [...prev, video];
+  //     }
+  //   });
+  // };
+
+  const addToHistory = async (item) => {
+    try {
+      const response = await axios.post(
```
Comment (m26unkwn) [inline src/context/history-context/history-context.js:47]: You can destruct the response.  `const {status, data:{history}} = api calls`

**Review by m26unkwn — APPROVED** [review]
Your Code is readable and clean.

File: src/context/history-context/history-context.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      if (response.status === 201) {
+        setHistory(response.data.history);
+        dispatch({
+          type: "TOAST_SUCCESS",
+          payload: "Added to History",
+        });
+      }
+    } catch (error) {
+      console.error(error.response.data.errors);
+      dispatch({ type: "TOAST_ERROR", payload: error.response.data.errors });
```
Comment (SinchanaMS) [inline src/context/history-context/history-context.js:65]: You could destructure the response.

File: src/context/history-context/history-context.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    }
+  };
+
+  // const deleteFromHistory = (video) => {
+  //   setHistory((prev) => prev.filter((v) => v._id !== video._id));
+  // };
+
```
Comment (SinchanaMS) [inline src/context/history-context/history-context.js:96]: Remove unused/commented code.

**Review by SinchanaMS — APPROVED** [review]

File: src/context/history-context/history-context.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    (async function () {
+      if (encodedToken) {
+        try {
+          const historyResponse = await axios.get("/api/user/history", {
+            headers: {
+              authorization: encodedToken,
+            },
+          });
+
+          if (historyResponse.status === 200) {
```
Comment (dev-ashishsethi) [inline src/context/history-context/history-context.js:25]: api success status can be between 200 and 299 make condition >200 and less than 299

File: src/context/history-context/history-context.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+      }
+    } catch (error) {
+      console.error(error.response.data.errors);
+      dispatch({ type: "TOAST_ERROR", payload: error.response.data.errors });
+    }
+  };
+
+  return (
+    <HistoryContext.Provider
+      value={{ history, addToHistory, clearHistory, deleteFromHistory }}
```
Comment (dev-ashishsethi) [inline src/context/history-context/history-context.js:119]: got a good insight on how to pass many states in value through provider thanks

**Review by dev-ashishsethi — APPROVED** [review]
do check comments

File: src/context/history-context/history-context.js (hunk 1 above)
Comment (manojsarna) [reply src/context/history-context/history-context.js:23]: Okay will check

File: src/context/history-context/history-context.js (hunk 5 above)
Comment (manojsarna) [reply src/context/history-context/history-context.js:25]: will look into this :)

**Pull request merged**